### PR TITLE
If MSVC, always link Arma dependencies

### PR DESCRIPTION
**Pull request by birm**
Following @rcurtin 's comment #2401 

I'm not familiar enough with whatever about windows caused #2401 to be sure that this would resolve it, but worth a try, right?

### CODE REVIEW

**Comment by rcurtin** [comment]
I don't have the right terminology and I'm only about 80% sure of what I'm saying here, but my understanding is that on Windows, we can't get away with "just" linking to the Armadillo library.  On Linux and Unix-like systems, the Armadillo library is linked to all of its dependencies (BLAS, LAPACK, ARPACK, whatever else...) and thus we only need to link against the Armadillo library.  But on Windows, this isn't true.  I can't find a reference for this, but that observation at least meshes with my experience.

Thus I *think*---but I'm not totally sure---that on Windows, we would want to link against Armadillo if `ARMA_USE_WRAPPER` is enabled, and *also* against all of its dependencies in all cases.  It seems like the code change here links against all dependencies on MSVC, but doesn't also add `ARMA_USE_WRAPPER`.  Do you think you could change it so it does both?  Or, if you know a reason why that's not the right thing to do, I'm all ears; like I said, I'm not totally sure. :)

**Comment by birm** [comment]
I _think_ this is what you mean. I should probably set up my own windows build environment to test various combinations, since I'm more or less doing this blind.

**Comment by rcurtin** [comment]
It might also be useful to add a comment explaining the rationale for the `OR MSVC` just in case someone comes along later and notices that the beliefs we wrote down are wrong. :smile:

File: CMake/FindArmadillo.cmake [hunk 1]
```diff
@@ -89,6 +89,9 @@ if(_ARMA_USE_WRAPPER)
 else()
   # don't link to armadillo in this case
   set(ARMADILLO_LIBRARY "")
+endif()
+# Link to support libraries in either case on MSVC
```
Comment (zoq) [inline CMake/FindArmadillo.cmake:93]:
```suggestion
# Link to support libraries in either case on MSVC.
```

Minor style issue.

**Pull request merged**